File: 0228-summary-ranges/0228-summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector <string> ans;
        if (!nums.size()) {
            return ans;
        }
        int n = nums.size();
        int lst = -1;
        for (int i = 0; i < n; i++) {
            if (lst == -1) {
                lst = i;
            } else if (0LL + nums[i] - nums[i - 1] > 1LL) {
                if (i - lst == 1) {
                    ans.push_back(to_string(nums[lst]));
                } else {
                    ans.push_back(to_string(nums[lst]) + "->" + to_string(nums[i - 1]));
                }
                lst = i;
            }
        }
        if (lst == n - 1) {
            ans.push_back(to_string(nums[lst]));
        } else {
            ans.push_back(to_string(nums[lst]) + "->" + to_string(nums.back()));
        }
        return ans;
    }
};

File: 0228-summary-ranges/0228-summary-ranges.go
func summaryRanges(nums []int) []string {
    if len(nums) == 0 {
        return []string{}
    }
    ans := make([]string, 0)
    start := nums[0]
    for i := 1; i < len(nums); i++ {
        if nums[i] != nums[i-1] + 1 {
            if nums[i-1] > start {
                ans = append(ans, fmt.Sprintf("%d->%d", start, nums[i-1]))
            } else {
                ans = append(ans, fmt.Sprintf("%d", start))
            }
            start = nums[i]
        }
    }
    if nums[len(nums)-1] > start {
        ans = append(ans, fmt.Sprintf("%d->%d", start, nums[len(nums)-1]))
    } else {
        ans = append(ans, fmt.Sprintf("%d", start))
    }
    return ans
}